core/templates/dev/head/components/HintAndSolutionButtonsDirective.ts

```diff
@@ -101,10 +101,11 @@ oppia.directive('hintAndSolutionButtons', [
             }
           };
 
-          ctrl.displaySolutionModal = function() {
-            ctrl.solutionModalIsActive = true;
-            var inPretestMode = ExplorationPlayerStateService.isInPretestMode();
-            if (!_editorPreviewMode && !inPretestMode) {
+          $scope.displaySolutionModal = function() {
```

Comment:
Do we need this function in scope?

Reply:
Sorry updated this, had some merging issues